pyccel/ast/variable.py

```diff
@@ -940,4 +940,4 @@ def __str__(self):
         return str(self.lhs)+'.'+str(self.name)
 
     def __repr__(self):
-        return '{}({}, dtype={})'.format(type(self).__name__, repr(self.lhs)+'.'+str(self.name), repr(self.dtype))
+        return f'{type(self).__name__}({repr(self.lhs)+'.'+str(self.name)}, dtype={repr(self.dtype)})'
```

Comment:
Beware. You have strings inside strings which is breaking your f-string into:
```
f'{type(self).__name__}({repr(self.lhs)+' \
.  \
'+str(self.name)}, dtype={repr(self.dtype)})'
```
You need to either vary the quotes used:
```
f"{type(self).__name__}({repr(self.lhs)+'.'+str(self.name)}, dtype={repr(self.dtype)})"
```
or use intermediate variables to make this clearer